### ISSUE TITLE
Support Back/Forward Mouse Buttons to navigate

### ISSUE THREAD

**Issue by flo2k** [mate-desktop/caja#78]
- The Back button should trigger the "Back" action (ALT+Left)
- The Forward button should trigger the "Forward" action (ALT+Right)

**Comment by jeffvd-a216** [comment]
Same. Just installed and upgraded Ubuntu MATE; I use list view quite a lot and like it, but I also like the mouse back and forward buttons for navigation. I came from Linux mint cinnamon.

**Comment by Stane1983** [comment]
I confirm issue as well on Mint 17.1. But, by accident I discovered that back and forward works just fine (in list view) when you have 2 or more tabs opened (ctrl+t) and your pointer is 'on tab', not inside listview. So my guess is if we can force tab part to be always visible it should work. Not a best solution but its solution.

**Comment by r3pek** [comment]
I can confirm the same situation and behavior as @Stane1983. Still, the best solution is for it to "just work (tm)"

**Comment by allanlaal** [comment]
issue seems to be fixed in MATE Desktop Environment 1.10.2

**Comment by r3pek** [comment]
I have MATE 1.10.2, Caja 1.10.4, but I still have the problem.

**Comment by allanlaal** [comment]
Weird, I have Caja 1.10.3

maybe its a library thing?

**Comment by ddska** [comment]
Ubuntu-MATE 15.10
MATE 1.10.2
Caja 1.10.4

Mouse "Back" button works only if I open Caja as administrator.

**Comment by zanzara** [comment]
Ubuntu-MATE 15.10
MATE 1.10.2
Caja 1.10.4

Mouse "Back" button works as user , but only in symbol view.  In list view NOT  :-( .
Pls. fix this issue for Mate 1.12

**Comment by raveit65** [comment]
Hmm, this works for me as user with caja-1.12, tested in Centos7.
Same with caja-1.8.2 with fedora-20.
Sorry, i don't have a VM with caja-1.10.x installed to confirm.
@flexiondotorg 
Can you check this?

**Comment by flexiondotorg** [comment]
@ddska Can you upgrade to MATE 1.12 and let us know if this is still an issue please? Use ppa:ubuntu-mate-dev/wily-mate

**Comment by dddux** [comment]
I have installed Ubuntu MATE and Mint MATE, both with MATE 1.12, and I have installed Debian Wheezy, and Jessie with MATE 1.8 and MATE 1.10.2 and Caja behaves the same in all of them regardless being a root or not. Back and Forward buttons only work in list view if you have open tabs and press the buttons on them. In the browser view they are registered as normal left button press. it would be really nice if you could squash this bug! ;) Thank you for the great work you've done with MATE! I love this DE!

**Comment by ddska** [comment]
@flexiondotorg on Mate 1.12.1 and Caja 1.12.3 situation is still the same, does not work.

**Comment by drws** [comment]
Status on Caja **1.12.6** and MATE **1.12.1**, running **as user**:

In _List View_: mouse buttons NOT working, not even when the mouse cursor is not hovering any icon

In _Icons/Compact View_: mouse buttons working ONLY when the mouse cursor is not over any icon - someone above said that it gets a while to get used to, but I think it should be also **considered as a bug** that the mouse back/forward buttons are not working while hovering over an icon (back and forward (hardware even) buttons have a clear functionality)

TL;DR: It is working as a user now (first bug resolved), but the functionality is still broken depending on Caja View mode (second and third bug unresolved).
